azure/components/component.ftl

```diff
@@ -1,5 +1,149 @@
 [#ftl]
 
+[#-- AutoScale --]
+[#assign autoScaleProfileChildrenConfiguration = 
```

Comment:
Is the autoscaling configuration we already have not sufficient?

Reply:
You're right - it should be sufficient.

What I've done so far does the job for Azure in a silo, but it not only introduces additional configuration it means that autoscaling approaches are completely different across providers. Ideally if you've worked in AWS with hamlet before and used an autoscaling component, then you should also know what to expect from the behaviour in Azure or any other provider.

I will take what I've implemented here and have it utilise the same stepped/tracked/scheduled approach that exists in the shared provider.

I'll update this PR to draft until I have that ready to go.